src/cli/analytics.rs

```diff
@@ -77,51 +84,148 @@ impl Command for Analytics {
     }
 }
 
+pub struct AnalyticsStream {
+    span: Span,
+    streams: StreamMap<String, RawJsonRowStreamer>,
+    // This allows us to extend the lifetime of the runtime used to create the streams longer than
+    // run(), else we panic when reading the streams
+    rt: Arc<Runtime>,
+}
+
+impl Iterator for AnalyticsStream {
+    type Item = Value;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some((cluster, result)) = self
+            .rt
+            .clone()
+            .block_on(async { self.streams.next().await })
+        {
+            let bytes = match result {
+                Ok(r) => r,
+                Err(e) => {
+                    return Some(Value::Error {
+                        error: Box::new(e),
+                        internal_span: self.span,
+                    });
+                }
+            };
+            let result_string = from_utf8(&bytes).unwrap();
+            let (start, _) = result_string.split_at(result_string.len() - 1);
+            let with_cluster = format!("{}, \"cluster\": \"{}\" }}", start, cluster);
+            let json_object = serde_json::from_str::<serde_json::Value>(&with_cluster).unwrap();
+            Some(convert_json_value_to_nu_value(&json_object, self.span).unwrap())
+        } else {
+            None
+        }
+    }
+}
+
 fn run(
     state: Arc<Mutex<State>>,
     engine_state: &EngineState,
     stack: &mut Stack,
     call: &Call,
     _input: PipelineData,
 ) -> Result<PipelineData, ShellError> {
-    let ctrl_c = engine_state.ctrlc.as_ref().unwrap().clone();
-    let statement: String = call.req(engine_state, stack, 0)?;
     let span = call.head;
 
     let cluster_identifiers = cluster_identifiers_from(engine_state, stack, &state, call, true)?;
 
     let guard = state.lock().unwrap();
 
+    let ctrl_c = engine_state.ctrlc.as_ref().unwrap().clone();
+    let statement: String = call.req(engine_state, stack, 0)?;
+
     let scope: Option<String> = call.get_flag(engine_state, stack, "scope")?;
-    let with_meta = call.has_flag(engine_state, stack, "with-meta")?;
+    let _with_meta = call.has_flag(engine_state, stack, "with-meta")?;
```

Comment:
Why is with_meta no longer used? If we don't want it to be used then should we remove it from the command?